docs/reference-implementation/Bring-your-own-workload.md

```diff
@@ -0,0 +1,80 @@
+
+- [Application Code](#application-code)
+- [Kubernetes](#kubernetes-deployment)
+- [Parameters](#parameters)
+
+## Application Code
+
+The simplest way to bring existing workloads into the reference implementation is to continue with the **mono-repo** approach. New workloads including their source code can be added to the repo, and the existing pipelines can be extended to build, push and deploy it to the infrastructure.
+
+In a **multi-repo** setup, the application code is stored in a different or in individual repositories, separated from each other and from the infrastructure. This also means that the application code is usually built and published with a separate pipeline and drop to shared location (e.g. Artifact store).
+
+The following two sub-sections contain information on how to either bring workloads using the mono-repo approach or the multi-repo approach into a cloned Azure Mission-Critical environment.
+
+### Mono-repo
+
+The application code is stored in the `/src/app/` directory. Each application has its own directory and contains a `Dockerfile` which is used to build the container image (for containerized workloads).
+
+```bash
+src/app/
+├── <application1>
+│   ├── Dockerfile
+│   ├── ...
+├── <application2>
+│   ├── ...
+```
+
+This `Dockerfile` is picked up by the `Build Application Code` stage of the pipeline. This pipeline can be easily extended to build and push additional images to the Azure Container Registry.
+
+The process itself is baked into a template that requires only three parameters:
+
+```yaml
+  - template: jobs-container-build.yaml
+    parameters:
+      jobName: '<app-name>' # unique pipeline job name
+      containerImageName: '<container-image-name>' # container image name
+      containerImageDockerFile: '<dockerfile>' # dockerfile used to build the container image
+```
+
+The [`jobs-container-build.yaml`](/.ado/pipelines/templates/jobs-container-build.yaml) template expects the files to be stored in `/src/app/`. The `containerImageDockerFile` parameter specifies the relative path from there to the `Dockerfile` i.e. `application1\Dockerfile`. This can be overwritten by the `workingDirectory` parameter if needed.
+
+### Multi-repo
+
+The main consideration here is to which registry the container images are pushed e.g. to a corporate-wide central container registry (see [Brownfield considerations](./Brownfield-Considerations.md)) or following the Azure Mission-Critical recommendation into a solution-specific one and how they can be accessed and pulled to the clusters we're deploying here.
```

Comment:
as below. Just dont make this a consideration but a straight recommendation: Do import into your own ACR.

Reply:
Agreed.

Reply:
Dropped.